core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -257,31 +238,27 @@ public void testMetricsForDecimals() throws IOException {
     record.setField("decimalAsInt64", new BigDecimal("4.75"));
     record.setField("decimalAsFixed", new BigDecimal("5.80"));
 
-    InputFile recordsFile = writeRecords(schema, record);
-
-    Metrics metrics = getMetrics(recordsFile);
+    Metrics metrics = getMetrics(schema, record);
     Assert.assertEquals(1L, (long) metrics.recordCount());
-    assertCounts(1, 1L, 0L, metrics);
+    assertCounts(1, 1L, 0L, 0L, metrics);
```

Comment:
Rather than changing so many rows that call `assertCounts`, you may want to consider adding a second variant of the method that accepts the count. Then you'd only need to change the double and float cases.

Reply:
I was thinking to make NaN count explicitly required so that people adding new tests will remember to take it into consideration, but I guess it's not needed at least before NaN is fully supported. Will update.

Reply:
I was thinking to make NaN count explicitly required so that people adding new tests will remember to take it into consideration, but I guess it's not needed at least before NaN is fully supported. Will update.